Lesson16/main.js

```diff
@@ -0,0 +1,149 @@
+const tabMenuList = document.getElementById("js-tab-menu__list");
+
+function addLoading() {
+  const tabContent = document.getElementById("js-tab-content");
+  const loading = document.createElement("img");
+  loading.src = "./loading-circle.gif";
+  loading.classList.add("loading");
+  loading.id = "loading";
+  tabContent.appendChild(loading);
+}
+
+function removeLoading() {
+  const loading = document.getElementById("loading");
+  loading.remove();
+}
+
+function addErrorMessage(error) {
+  const tabContent = document.getElementById("js-tab-content");
+  const errorMessage = document.createElement("p");
+  errorMessage.classList.add("error-message");
+  errorMessage.textContent = error;
+  tabContent.appendChild(errorMessage);
+}
+
+async function fetchErrorHandling(response) {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error("サーバーエラーが発生しました");
+  }
+}
+
+async function getJsonOrError(url) {
+  const response = await fetch(url);
+  const json = await fetchErrorHandling(response);
+  return json;
+}
+
+async function getArrayFetchData() {
+  const resource = [
+    "./news.json",
+    "./book.json",
+    "./travel.json",
+    "./economy.json"
+  ];
+  try {
+    const data = await Promise.all(resource.map(getJsonOrError));
+    return data.filter((value) => value !== undefined);
+  } catch (e) {
+    addErrorMessage(e);
+  }
+}
+
+function createTabMenu(data) {
+  for (let i = 0; i < data.length; i++) {
+    const tabMenuItem = document.createElement("li");
+    tabMenuItem.classList.add("tab-menu__item");
+    tabMenuItem.id = data[i].category;
+    tabMenuItem.textContent = data[i].category;
+    tabMenuList.appendChild(tabMenuItem);
+  }
+}
+
+function createTabContent() {
+  const tabContent = document.createElement("div");
+  const imageWrapper = document.createElement("div");
+  const tabContentList = document.createElement("ul");
+
+  tabContent.id = "js-tab-content";
+  tabContentList.id = "js-tab-content__list";
+  imageWrapper.id = "js-img-wrapper";
+
+  imageWrapper.classList.add("tab-content__img-wrapper");
+  tabContent.classList.add("tab-content");
+
+  tabContent.appendChild(tabContentList);
+  tabContent.appendChild(imageWrapper);
+  tabMenuList.parentNode.insertBefore(tabContent, tabMenuList.nextSibling);
+}
+
+async function configUIfromFetchData() {
+  const data = await getArrayFetchData();
+  if (data) {
+    createTabMenu(data);
+    const hasSelectData = data.find((value) => value.select === true);
+    const tab = document.getElementById(hasSelectData.category);
+    tab.classList.add("active");
+    createElement(hasSelectData);
+    addImage(hasSelectData);
+  }
+}
+
+async function createElement({ article }) {
```

Comment:
nis
I know what the other functions are just from reading them, so it would be nice to know more about this one too, but it's a minor detail, so I'll leave it up to you to fix it!

```suggestion
async function createArticleElements.({ article }) {
```

Reply:
Thanks for the review.
I agreed with your opinion, so...
I will make the correction.
98039aa